### ISSUE TITLE
Using HasExited and ExitCode layout renderers

### ISSUE THREAD

**Issue by nirajdoshi1985** [NLog/NLog#979]
I am using a nlog 4 logger for my WPFapplication (but it can be any .net application). In NLog Documentation there are several parameter options available for {processinfo} layout renderer. Out of them I am trying to use {processinfo:property=HasExited} and {processinfo:property=ExitCode}.

Now for example my main application has ProcessName:SampleApplication.vshost. How can I log ExitCode and HasExit for my current application. Because even in OnExit event of Application, the application has still not exited.

Or for e.g. if I do Process.Start("Calculator.exe") and if at some point it is closed how can a logger log Calculators exit code, specifically using ${processinfo:property=HasExited} and ${processinfo:property=ExitCode}

So basically, my question is how to utilize these layout parameters. I am not asking about solutions like handling OnExit event of Application and logging the ExitEventArgs.ApplicationExitCode.

**Comment by 304NotModified** [comment]
Hi,

I'm not sure how to answer your question, but things good to know:
We use `Process.GetCurrentProcess();` and `Process.HasExited`/ `ExitCode`.

But I think that can't be working with your current process....

**Comment by 304NotModified** [comment]
We need to add this to the docs.

**Comment by snakefoot** [comment]
Sounds strange to log the ExitCode from `Process.GetCurrentProcess()`. Any way updated the wiki: https://github.com/NLog/NLog/wiki/ProcessInfo-Layout-Renderer
